### PR TITLE
Replaces XOR with AES-GCM for master seed and pubkey protection for TARGET=dev

### PR DESCRIPTION
**Pull request by syncom**
We integrate Brian Gladman's AES GCM implementation in his [modes](https://github.com/BrianGladman/modes) github repo in subzero, and use it to protect master seeds and public keys for the dev target.

This change makes subzero's dev and ncipher targets one step closer to each other. It also makes it easier to have unified regression test vectors for both targets.

This pull request contains two commits: The first one integrates Brian Gladman's GCM code in `trezor-crypto/`; the second one uses it to protect master seeds and public keys in the dev target. It is not ideal to introduce foreign code in `trezor-crypto`, to make it deviate further from upstream. However, this seems to be the more logical and straightforward approach from the point of view of source code organization, and therefore I've decided to go with it.

Self tests have passed for the dev build.

### CODE REVIEW

**Comment by alokmenghrajani** [comment]
I don't think you should put foreign code in trezor-crypto. Just keep it in another folder (or get the change accepted upstream).

**Comment by syncom** [comment]
> I don't think you should put foreign code in trezor-crypto. Just keep it in another folder (or get the change accepted upstream).
Thanks for the feedback, @alokmenghrajani. I will put it in a separate folder then. Because aes-gcm might not be essential for `trezor-crypto` (although we use it in the HSM), getting it accepted upstream might not be a high priority (for both us and upstream).

**Comment by syncom** [comment]
Took a tiny subset of Brian Gladman's AES and Modes code for AES GCM support. We have copy-and-pasted Gladman's github code AS IS to the `external-crypto` directory, and maintain a CMakeLists.txt file for building GCM.

File: core/CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,7 @@ if ($ENV{TARGET} MATCHES "nCipher")
 endif ()
 
 add_subdirectory(trezor-crypto)
+add_subdirectory(external-crypto)
```
Comment (oreparaz) [inline core/CMakeLists.txt:19]: 👌🏻

File: core/trezor-crypto/CMakeLists.txt [hunk 2]
```diff
@@ -28,6 +28,7 @@ target_include_directories(TrezorCrypto PUBLIC chacha20poly1305)
 target_include_directories(TrezorCrypto PUBLIC ed25519-donna)
 
 target_compile_options(TrezorCrypto PRIVATE "-std=c99")
+add_definitions(-Wno-cast-qual -Wno-incompatible-pointer-types -Wno-missing-prototypes)
```
Comment (oreparaz) [inline core/trezor-crypto/CMakeLists.txt:31]: does this apply to trezor crypto or external crypto?

**Review by oreparaz — APPROVED** [review]

File: core/trezor-crypto/CMakeLists.txt (hunk 2 above)
Comment (syncom) [reply core/trezor-crypto/CMakeLists.txt:31]: It applies to the former. Note that this is the cmake file for trezor-crypto.

**Review by oreparaz — APPROVED** [review]

**Comment by syncom** [comment]
Once merged, this PR will fail the signtx regression test, due to the fact that an older wallet (protected with XOR) is used for the test. This is expected, and will be resolved by a follow-up PR. FYI, @oreparaz

**Pull request merged**